third_party/nvidia/lib/Dialect/NVWS/Transforms/LowerAref.cpp

```diff
@@ -70,14 +71,36 @@ void assignStageCluster(Operation *op, std::optional<PartitionSet> partitionIds,
                         StageCluster stageCluster, OpBuilder &builder) {
   if (partitionIds) {
     setPartition(op, *partitionIds);
+    setStageCluster(builder, op, stageCluster);
+  }
+}
+
+bool isOperandPipelineable(Value v, scf::ForOp forOp) {
+  auto isPipelineable = [](Operation *op) {
+    return isa<ArefPutEnterOp, ArefGetEnterOp, ArefBufferOp>(op);
+  };
 
-    if (stageCluster) {
-      op->setAttr(triton::kLoopStageAttrName,
-                  builder.getI32IntegerAttr(stageCluster->first));
-      op->setAttr(triton::kLoopClusterAttrName,
-                  builder.getI32IntegerAttr(stageCluster->second));
+  Operation *foundDef = nullptr;
+  return triton::nvidia_gpu::isOperandPipelineableBase(v, forOp, foundDef,
+                                                       isPipelineable);
```

Comment:
Why is it ok to omit `isLoadToBePipelined` for WS? Because the load is manually pipelined anyway?